flake.nix

```diff
@@ -1,77 +1,128 @@
 {
-  description = "Template for Holochain app development";
+  description = "Flake for Holochain app development";
 
   inputs = {
-    nixpkgs.follows = "holochain/nixpkgs";
-    versions.url = "github:holochain/holochain?dir=versions/0_3";
+    holonix.url = "github:holochain/holonix?ref=main-0.3";
 
-    holochain = {
-      url = "github:holochain/holochain";
-      inputs.versions.follows = "versions";
+    crane.url = "github:ipetkov/crane";
+
+    nixpkgs.follows = "holonix/nixpkgs";
+    flake-parts.follows = "holonix/flake-parts";
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "holonix/nixpkgs";
```

Comment:
```suggestion
nixpkgs.url = "github:nixos/nixpkgs?ref=24.05";
flake-parts.url = "github:hercules-ci/flake-parts";
holonix = {
    url = "github:holochain/holonix?ref=main-0.3";
    inputs = {
        nixpkgs.follows = "nixpkgs";
        flake-parts.follows = "flake-parts";
        crane.follows = "crane";
        rust-overlay.follows = "rust-overlay";
    };
};

crane.url = "github:ipetkov/crane";

rust-overlay = {
      url = "github:oxalica/rust-overlay";
      inputs.nixpkgs.follows = "nixpkgs";
};

```

Reply:
Something like this, so that your flake controls the inputs. It will affect caching if they get too far behind but it means you get more stability and don't risk having duplicate copies of dependencies and getting strange errors